lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -69,27 +52,18 @@ def find_first_by_auth_conditions(tainted_conditions, opts = {})
 
           if email && email.is_a?(String)
             conditions = filtered_conditions.to_h.merge(opts).
-              reverse_merge(emails: { email: email })
+              reverse_merge(multi_email_association.reflection.table_name => { email: email })
```

Comment:
Further tested the table name approach on our APP and it also works (haven't managed to find a case that association name does not work but table name works though). 

Given the issue that you posted, let's stick with this then, thank you !